rest-service/manager_rest/storage/storage_manager.py

```diff
@@ -178,11 +178,14 @@ def _add_permissions_filter(self, query, model_class):
         if not has_request_context():
             return query
 
-        # For users that are allowed to see all resources, regardless of tenant
         # Queries of elements that aren't resources (tenants, users, etc.),
-        # shouldn't be filtered as well
+        # shouldn't be filtered
+        if not model_class.is_resource:
+            return query
+
+        # For users that are allowed to see all resources, regardless of tenant
         is_admin = is_administrator(self.current_tenant)
-        if not model_class.is_resource or is_admin:
+        if is_admin:
```

Comment:
At this point, might as well not split this into 2 lines, but do `if is_administrator(self.current_tenant)`